src/deadline_test_fixtures/deadline/resources.py

```diff
@@ -332,7 +393,17 @@ class Job:
     template: dict
 
     name: str
-    lifecycle_status: str
+    lifecycle_status: Literal[
```

Comment:
Should we change this to use the `JobLifecycleStatus` enum?

Reply:
Discussed offline and the one concern is that this PR becomes a breaking change. We decided that this was an overall improvement valuable enough to justify the breaking change. Rationale:

1.  Using an enum will help test authors with IDEs that support type hints and code completion. Supporting IDEs will suggest the possible values for the enum making it easier to write tests.
2. We are pre-`1.0` and we believe that there are very few consumers of this package &mdash; likely only consumed by AWS Deadline Cloud repositories.

I've switched to the enum and marked the commit as a breaking change